distribution/registry.go

```diff
@@ -60,14 +60,17 @@ func NewV2Repository(ctx context.Context, repoInfo *registry.RepositoryInfo, end
 	endpointStr := strings.TrimRight(endpoint.URL, "/") + "/v2/"
 	req, err := http.NewRequest("GET", endpointStr, nil)
 	if err != nil {
-		return nil, false, err
+		return nil, false, false, err
```

Comment:
@stevvooe: I've updated the PR to use a different approach, please tell me if you think it's an improvement. Now `NewV2Repository` returns a `fallbackError` directly instead of depending on the caller to wrap the error it returns. This means we can skip the extra return value.